### PR TITLE
Python38

### PR DESCRIPTION
**Pull request by FlorianLudwig**
rebased version of https://github.com/beerfactory/hbmqtt/pull/226

depends on https://github.com/Yakifo/aio-hbmqtt/pull/17

### CODE REVIEW

File: tests/plugins/test_authentication.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def test_allow_anonymous(self):
         s.username = ""
         auth_plugin = AnonymousAuthPlugin(context)
         ret = self.loop.run_until_complete(auth_plugin.authenticate(session=s))
-        self.assertTrue(ret)
+        assert ret
```
Comment (HerrMuellerluedenscheid) [inline tests/plugins/test_authentication.py:33]: Wouldn't it be better to `assert is True`? What if `auth_plugin.authenticate` changes with regard to the returned value?
Same for lines 61 and 81.

**Comment by HerrMuellerluedenscheid** [comment]
I love where this is going! Thanks for addressing the tests @FlorianLudwig!

File: tests/plugins/test_authentication.py (hunk 1 above)
Comment (FlorianLudwig) [reply tests/plugins/test_authentication.py:33]: I prefer to not spread the typing information all over the tests but if we want to be strict in regards to types we should add typing hints to the corresponding function, e.g. `-> bool` to `authenticate` and include a type checker in our tests

Comment (HerrMuellerluedenscheid) [reply tests/plugins/test_authentication.py:33]: Sounds good. Type hinting and `mypy` should be the way to go here.

**Comment by HerrMuellerluedenscheid** [comment]
@FlorianLudwig do we want to squash commits? I could squash all of mine into a single `python3 refactor` I guess. But I don't want to do that without coordinating with you to avoid conflicts. If you consider squashing any of your feel free to do the same for mine. But after all it's just cosmetic. So, what do you think about squashing?

**Comment by FlorianLudwig** [comment]
# tl;dr
pragmatic, i would just leave it as is.

# squashing commits
It is on my TODO list to write my thoughts about squashing - a complicated topic indeed. So let me have a take on it :) Just to make a simple question a discussion about principles.

The git history is part of the work of a developer and not a backup. It should tell a story, each commit is one scene. Jumping to much between them and you lose your reader (that might be yourself in a few month, trying to understand why the code ended up the way it did or some [psychopath](http://ruby.zigzo.com/2014/08/01/who-said-that-one-violent-psychopath-quote/)).

Some rules of thumb:

 * One commits should not never include unrelated changes
 * Generally, try to keep commits small, making it easier to review each commit on it's own
 * Automated changes to files that do not change the meaning (e.g. reformatting) must be a separate commit with the tools used clearly stated in the commit message
 * Often before doing the change you actually want to do, there is some refactoring involved, in that case: 
 
   Move refactoring into its own commit.
   A refactoring should keep the same end result of the software, while changing the code.
   Having a refactoring separated into a separate commit from other changes keeps them comprehensible and reviewable.

**Review by HerrMuellerluedenscheid — APPROVED** [review]
